compat/maven-compat/src/main/java/org/apache/maven/repository/metadata/MetadataGraphEdge.java

```diff
@@ -30,9 +30,9 @@
 public class MetadataGraphEdge {
     String version;
     ArtifactScopeEnum scope;
-    int depth = -1;
-    int pomOrder = -1;
-    boolean resolved = true;
+    int depth;
```

Comment:
removing those is wrong. Currently, calling "getPomOrder()" in a subclass constructor (which is called from super()) will return -1. After this change, it will return 0. 

This is not just a simple cleanup, but changing the actual logic of the class.